src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,815 @@
+                self.is_processing = False  # Mark that processing is complete
+
+    def _do_correlation(self):
+        """Run the correlation between the FIB and FM images."""
+        # Modify the input data to match the required format to run 3DCT
+        fm_das = [stream.raw[0] for stream in self.correlation_target.fm_streams]
+        fm_image = _convert_das_to_numpy_stack(fm_das)
+        fib_da = self.correlation_target.fib_stream.raw[0]
+        fib_coords = []
+        fm_coords = []
+        poi_coords = []
+        path = os.getcwd()
+        for fib_coord in self.correlation_target.fib_fiducials:
+            fib_coord = self.correlation_target.fm_streams[0].getPixelCoordinates_alt(fib_coord.coordinates.value[0:2])
+            fib_coords.append(fib_coord)
+        fib_coords = numpy.array(fib_coords, dtype=numpy.float32)
+        for fm_coord in self.correlation_target.fm_fiducials:
+            fm_coord_2d = self.correlation_target.fm_streams[0].getPixelCoordinates_alt(fm_coord.coordinates.value[0:2])
+            fm_coords.append([fm_coord_2d[0], fm_coord_2d[1], fm_coord.coordinates.value[2]])
+        fm_coords = numpy.array(fm_coords, dtype=numpy.float32)
+        min_n = min(len(fib_coords), len(fm_coords)) # make fm coords and fib coords the same length
+        fib_coords = fib_coords[:min_n]
+        fm_coords = fm_coords[:min_n]
+        poi_coord = self.correlation_target.fm_pois
+        poi_coord_2d = self.correlation_target.fm_streams[0].getPixelCoordinates_alt(poi_coord.coordinates.value[0:2])
+        poi_coords.append([poi_coord_2d[0], poi_coord_2d[1], poi_coord.coordinates.value[2]])
+        poi_coords = numpy.array(poi_coords, dtype=numpy.float32)
+        # Run the correlation
+        self.correlation_target.correlation_result = run_tdct_correlation(fib_coords=fib_coords, fm_coords=fm_coords,
+                                                                          poi_coords=poi_coords,
+                                                                          fib_image=fib_da, fm_image=fm_image,
+                                                                          path=path)
+        # Update the output pameters of the correlation
+        self._tab_data_model.projected_points = []
+        self.correlation_target.fib_projected_fiducials = []
+        points = self.correlation_target.correlation_result['output']['error']['reprojected_3d']
+        for n, i in enumerate(points[0]):
+            p_pos = self.correlation_target.fib_stream.getPhysicalCoordinates((points[0][n], points[1][n]))
+            target = Target(x=p_pos[0], y=p_pos[1], z=0, name="PP" + str(n + 1), type="ProjectedPoints", index=n + 1,
+                            fm_focus_position=0)
+            self._tab_data_model.projected_points.append(target)
+            self.correlation_target.fib_projected_fiducials.append(target)
+
+        projected_poi = get_reprojected_poi_coordinate(self.correlation_target.correlation_result)
```

Comment:
I think this might be one of the correlation errors, this projected_poi in in metres from center of image, don't think it can be directly used like this. You will probably need one of the other poi results image_px, or px